### PR TITLE
reject connections if backends are down

### PR DESCRIPTION
**Pull request by aojea**

### CODE REVIEW

File: pkg/loadbalancer/proxy.go [hunk 1]
```diff
@@ -61,6 +61,8 @@ defaults
 frontend {{$index}}-frontend
   bind {{ $data.BindAddress }}
   default_backend {{$index}}-backend
+  # reject connections if all backends are down
+  tcp-request connection reject if { nbsrv({{$index}}-backend) lt 1 }
```
Comment (BenTheElder) [inline pkg/loadbalancer/proxy.go:65]: https://www.haproxy.com/documentation/haproxy-configuration-manual/latest/#4-tcp-request%20connection

Comment (BenTheElder) [reply pkg/loadbalancer/proxy.go:65]: so this is basically https://stackoverflow.com/a/17712250 :-)

Comment (aojea) [reply pkg/loadbalancer/proxy.go:65]: is the only test failing in CI :) https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/124117/pull-kubernetes-e2e-kind-cloud-provider-loadbalancer/1774932103805276160

**Comment by aojea** [comment]
/hold cancel

the existing e2e test with bash is very flaky, I'm going to remove it too

File: pkg/loadbalancer/proxy.go (hunk 1 above)
Comment (BenTheElder) [reply pkg/loadbalancer/proxy.go:65]: that's awesome!

**Pull request merged**